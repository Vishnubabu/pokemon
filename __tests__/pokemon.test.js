/**
 * @jest-environment jsdom
 */
import { render, cleanup, waitFor } from "@testing-library/react";

// Mock useRouter
jest.mock("next/navigation", () => ({
  useRouter() {
    return {
      replace: () => null
    };
  }
}));

// Mock Chart
jest.doMock('react-apexcharts', () => () => null);

import Page from "@/app/pokemon/[...slug]/page";

function mockFetch(data) {
  return jest.fn().mockImplementation(() =>
    Promise.resolve({
      ok: true,
      json: () => data,
    }),
  );
}

afterEach(cleanup);

describe("Pokemon page", () => {
  it("renders pokemon details", async () => {
    window.fetch = mockFetch({
      "abilities": [
        {
          "ability": {
            "name": "rock-head",
          }
        }
      ],
      "forms": [
        {
          "name": "golem",
        }
      ],
      "height": 14,
      "held_items": [
        {
          "item": {
            "name": "hard-stone",
          },
        }
      ],
      "name": "golem",
      "species": {
        "name": "golem",
      },
      "stats": [
        {
          "base_stat": 80,
          "stat": {
            "name": "hp",
          }
        },
      ],
      "types": [
        {
          "type": {
            "name": "rock",
            "url": "https://pokeapi.co/api/v2/type/6/"
          }
        }
      ],
      "weight": 3000
    });

    const { container, getByTestId } = render(<Page params={{ slug: ['golem'] }} />);
    await waitFor(() => {
      expect(getByTestId('list')).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });
});
