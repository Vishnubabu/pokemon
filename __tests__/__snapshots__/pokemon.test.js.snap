// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon page renders pokemon details 1`] = `
<div>
  <h1
    class="mb-4 text-xl font-extrabold md:text-3xl lg:text-4xl"
  >
    <span
      class="text-transparent bg-clip-text bg-gradient-to-r to-emerald-600 from-sky-400 capitalize"
    >
      golem
    </span>
     
  </h1>
  <img
    alt="golem"
    height="96"
    loading="lazy"
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/undefined.png"
    width="96"
  />
  <div
    class="mt-4 w-full max-w-md p-4 bg-white border border-gray-200 rounded-lg shadow sm:p-8"
  >
    <div
      class="flex items-center justify-between mb-4"
    >
      <h2
        class="text-xl font-bold leading-none"
      >
        Information
      </h2>
    </div>
    <div
      class="flow-root"
    >
      <ul
        class="divide-y divide-gray-200"
        data-testid="list"
        role="list"
      >
        <li
          class="py-3 sm:py-4"
        >
          <div
            class="flex items-center justify-between"
          >
            <p
              class="text-sm font-medium truncate"
            >
              Species
            </p>
            <div
              class="capitalize inline-flex items-center text-base text-right font-semibold"
            >
              golem
            </div>
          </div>
        </li>
        <li
          class="py-3 sm:py-4"
        >
          <div
            class="flex items-center justify-between"
          >
            <p
              class="text-sm font-medium truncate"
            >
              Height
            </p>
            <div
              class="inline-flex items-center text-base text-right font-semibold"
            >
              1.4 m
            </div>
          </div>
        </li>
        <li
          class="py-3 sm:py-4"
        >
          <div
            class="flex items-center justify-between"
          >
            <p
              class="text-sm font-medium truncate"
            >
              Weigth
            </p>
            <div
              class="inline-flex items-center text-base text-right font-semibold"
            >
              300 kg
            </div>
          </div>
        </li>
        <li
          class="py-3 sm:py-4"
        >
          <div
            class="flex items-center justify-between"
          >
            <p
              class="text-sm font-medium truncate"
            >
              Abilities
            </p>
            <div
              class="capitalize inline-flex items-center text-base text-right font-semibold"
            >
              rock head
            </div>
          </div>
        </li>
        <li
          class="py-3 sm:py-4"
        >
          <div
            class="flex items-center justify-between"
          >
            <p
              class="text-sm font-medium truncate"
            >
              Forms
            </p>
            <div
              class="capitalize inline-flex items-center text-base text-right font-semibold"
            >
              golem
            </div>
          </div>
        </li>
        <li
          class="py-3 sm:py-4"
        >
          <div
            class="flex items-center justify-between"
          >
            <p
              class="text-sm font-medium truncate"
            >
              Held Items
            </p>
            <div
              class="capitalize inline-flex items-center text-base text-right font-semibold"
            >
              hard stone
            </div>
          </div>
        </li>
        <li
          class="py-3 sm:py-4"
        >
          <div
            class="flex items-center justify-between"
          >
            <p
              class="text-sm font-medium truncate"
            >
              Categories
            </p>
            <div
              class="flex gap-2 inline-flex items-center text-base text-right font-semibold"
            >
              <a
                class="capitalize hover:underline inline-flex items-center"
                href="/category/rock/6"
              >
                rock
                <svg
                  class="w-4 h-4 ms-2 rtl:rotate-180"
                  fill="none"
                  viewBox="0 0 14 10"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1 5h12m0 0L9 1m4 4L9 9"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </a>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="mt-4 w-full max-w-md p-4 bg-white border border-gray-200 rounded-lg shadow sm:p-8"
  >
    <div
      class="flex items-center justify-between mb-4"
    >
      <h2
        class="text-xl font-bold leading-none"
      >
        Stats
      </h2>
    </div>
    <div
      class="flow-root"
    />
  </div>
</div>
`;
